diff --git a/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/CameraSession.kt b/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/CameraSession.kt
index 92c65a0..b59801f 100644
--- a/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/CameraSession.kt
+++ b/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/CameraSession.kt
@@ -302,6 +302,7 @@ class CameraSession(private val context: Context, private val cameraManager: Cam
         size.width,
         size.height,
         video.config.pixelFormat,
+        Orientation.fromRotationDegrees(characteristics.get(CameraCharacteristics.SENSOR_ORIENTATION) ?: 0),
         isSelfie,
         video.config.enableFrameProcessor
       )
diff --git a/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/VideoPipeline.kt b/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/VideoPipeline.kt
index 77440b7..5f5d07c 100644
--- a/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/VideoPipeline.kt
+++ b/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/core/VideoPipeline.kt
@@ -28,6 +28,7 @@ class VideoPipeline(
   val width: Int,
   val height: Int,
   val format: PixelFormat = PixelFormat.NATIVE,
+  private val orientation: Orientation,
   private val isMirrored: Boolean = false,
   enableFrameProcessor: Boolean = false
 ) : SurfaceTexture.OnFrameAvailableListener,
@@ -106,7 +107,7 @@ class VideoPipeline(
         val image = reader.acquireNextImage() ?: return@setOnImageAvailableListener
 
         // TODO: Get correct orientation and isMirrored
-        val frame = Frame(image, image.timestamp, Orientation.PORTRAIT, isMirrored)
+        val frame = Frame(image, image.timestamp, orientation, isMirrored)
         frame.incrementRefCount()
         frameProcessor?.call(frame)
 
diff --git a/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/extensions/CameraDevice+createPhotoCaptureRequest.kt b/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/extensions/CameraDevice+createPhotoCaptureRequest.kt
index 0c425a8..5e66a7d 100644
--- a/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/extensions/CameraDevice+createPhotoCaptureRequest.kt
+++ b/node_modules/react-native-vision-camera/android/src/main/java/com/mrousavy/camera/extensions/CameraDevice+createPhotoCaptureRequest.kt
@@ -51,7 +51,7 @@ fun CameraDevice.createPhotoCaptureRequest(
   }
   captureRequest.set(CaptureRequest.JPEG_QUALITY, jpegQuality.toByte())
 
-  captureRequest.set(CaptureRequest.JPEG_ORIENTATION, orientation.toDegrees())
+  captureRequest.set(CaptureRequest.JPEG_ORIENTATION, 0)
 
   // TODO: Use the same options as from the preview request. This is duplicate code!
 
